### PR TITLE
Search/Alerts/Auth

### PR DESCRIPTION
**Pull request by itisdb**
- Search bar has been implemented in the front end
- the is_authenticated() function has been used in App.vue
- alerts integration has been done.

If you find any disarray in the code or a snippet that you don't understand please let me know.

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -1,33 +1,37 @@
...
+    },
   },
-  watch:{
-    $route (){
-      this.$store.dispatch('alertStore/clear');
-    }
+  watch: {
+    $route() {
+      this.$store.dispatch("alertStore/clear");
+    },
```
Comment (chayan007) [inline src/App.vue:34]: erase trailing `,` if not required

File: src/components/NavReg.vue [hunk 2]
```diff
@@ -38,25 +38,32 @@
             </ul>
           </li>
 
-<!-- this is just a dev component this will be removed-->
-
+          <!-- this is just a dev component this will be removed-->
```
Comment (chayan007) [inline src/components/NavReg.vue:41]: Please give proper name to components. `NavRegistration` or `NavAuthMenu`

File: src/App.vue [hunk 3]
```diff
@@ -1,33 +1,37 @@
 <template>
-  <div v-if="true"><Nav/></div>
-  <div v-else><NavReg/></div>
+  <div v-if="is_authenticated()"><Nav /></div>
+  <div v-else><NavReg /></div>
 
-  <div v-if="alert.message" :class="`alert ${alert.type}`">{{alert.message}}</div>
+  <div v-if="alert.message" :class="`alert ${alert.type}`">
+    <Alert />
```
Comment (chayan007) [inline src/App.vue:6]: How are you getting the message in the component ? Props from this component has not been passed onto this child component `Alert`.

File: src/router/index.js [hunk 4]
```diff
@@ -24,12 +26,28 @@ const routes = [
   {
     path: "/",
     name: "Home",
-    component: Home
-  },
+    get component() {
+      const x=1; // emulating the value of the condition check for the sign in.
```
Comment (chayan007) [inline src/router/index.js:30]: Please remove component switching logic based on authentication from routes to the view.

File: src/views/search.vue [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+</template>
+
+<script>
+import SearchBar from '../components/search/searchBar.vue'
+
+
+export default {
+    name: "Search",
+    components:{SearchBar
+
```
Comment (chayan007) [inline src/views/search.vue:20]: Please check these lints of trailing spaces and `,`

File: src/views/search.vue [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+            <div class="col text-center mt-5">
+                    <span class="h2 text-center">Search results:</span>
+            </div>
+        </div>
+        <SearchBar/>
+            </div></section>
+</template>
+
+<script>
+import SearchBar from '../components/search/searchBar.vue'
```
Comment (chayan007) [inline src/views/search.vue:14]: use `@` notation

File: src/router/index.js [hunk 7]
```diff
@@ -3,7 +3,9 @@ import SignUp from '../components/SignUp.vue'
 import Register from '../views/Register.vue'
 import Login from '../views/Login.vue'
 import Home from '../views/Home.vue'
+import Splash from '../views/Splash.vue'
```
Comment (chayan007) [inline src/router/index.js:6]: use `@` notation for all.

File: src/components/search/searchBar.vue [hunk 8]
```diff
@@ -0,0 +1,26 @@
+<template>
+    <div
+        class="input-group
+            "
```
Comment (chayan007) [inline src/components/search/searchBar.vue:4]: Please see these lint issues.

File: src/components/search/searchBar.vue [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+        />
+        <div class="input-group-append">
+          <span class="input-group-text"
+            ><router-link to="/search"><span class="fas fa-search"></span
+          ></router-link></span>
+        </div>
+      </div>
+</template>
+<script>
+//import { defineComponent } from '@vue/composition-api'
```
Comment (chayan007) [inline src/components/search/searchBar.vue:21]: If not required, remove completely.

File: src/components/alerts/alert.vue [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+    <span class="alert-inner--text"
+      ><strong>Heads up!</strong> {{ alert.message }}
+    </span>
+    <button type="button" class="close" data-dismiss="alert" aria-label="Close">
+      <span aria-hidden="true">×</span>
+    </button>
+  </div>
+</template>
+<script>
+export default {
```
Comment (chayan007) [inline src/components/alerts/alert.vue:50]: Can you please attach screenshots of this component ?

File: src/components/NavReg.vue [hunk 11]
```diff
@@ -38,25 +38,32 @@
...
+              </li>
+              <li>
+                <a class="dropdown-item"
+                  ><router-link to="/register">Register</router-link></a
+                >
+              </li>
+              <li>
+                <a class="dropdown-item"
+                  ><router-link to="/login">Login</router-link></a
+                >
```
Comment (chayan007) [inline src/components/NavReg.vue:61]: Please check lints for this entire file properly.

**Review by chayan007 — CHANGES_REQUESTED** [review]

File: src/App.vue (hunk 1 above)
Comment (itisdb) [reply src/App.vue:34]: ok working on it

File: src/components/Navigation.vue [hunk 12]
```diff
@@ -33,7 +33,11 @@
               <span class="nav-link-inner-text">Menu</span>
             </a>
             <ul class="dropdown-menu">
-              <li><a class="dropdown-item"><router-link to='/settings'>Settings</router-link></a></li>
+              <li>
+                <a class="dropdown-item"
+                  ><router-link to="/settings">Settings</router-link></a
+                >
```
Comment (chayan007) [inline src/components/Navigation.vue:39]: @itisdb please try to check this kind of issues. Maybe you are using some kind of prettier formatter which is creating such messy code.

WRONG:
```
...</a
>
```

CORRECT:
```
...
</a>
```

File: src/components/SettingsPage.vue [hunk 13]
```diff
@@ -251,7 +251,7 @@
 </template>
 
 <script>
-import AdSettings from "@/components/adset.vue";
+import AdSettings from "@/components/AdvancedSettings.vue";
 export default {
   name: "SettingsPage",
   components: { AdSettings },
```
Comment (chayan007) [inline src/components/SettingsPage.vue:257]: Please rename this to `AdvancedSettings`, there will be an `AdSettings` component in future.

File: src/components/search/searchBar.vue (hunk 8 above)
Comment (chayan007) [reply src/components/search/searchBar.vue:4]: @itisdb please check this lint issues. One line is used just for `"`

File: src/views/search.vue (hunk 6 above)
Comment (chayan007) [reply src/views/search.vue:14]: @itisdb

File: src/components/Navigation.vue [hunk 14]
```diff
@@ -121,29 +132,82 @@
...
-      >
-        <img
-          src="../assets/profile-picture-1.jpg"
-          class="card-img-top rounded-circle"
-          alt="profile image"
-        />
+      <!--search bar button invoking the search page-->
+      
+
+
```
Comment (chayan007) [inline src/components/Navigation.vue:138]: Please check, there are lot of wasted newlines.

File: src/components/Navigation.vue (hunk 12 above)
Comment (chayan007) [reply src/components/Navigation.vue:39]: @itisdb unaddressed

**Review by chayan007 — APPROVED** [review]

**Review by vikas339 — APPROVED** [review]

**Pull request merged**